CARDS-1415: User dashboard and side bar terminology improvements

marta- opened the pull request:
Includes:
- [x] [CARDS-1434](https://phenotips.atlassian.net/browse/CARDS-1434) _"Subjects" box improvements_:
* Removed the “Subjects” title
* Moved the tabs up instead of the title, and used title font
* Made all subject type labels plural
- [x] [CARDS-1438](https://phenotips.atlassian.net/browse/CARDS-1438) _"Forms" box improvements_ and
- [x] [CARDS-1370](https://phenotips.atlassian.net/browse/CARDS-1370) _In addition to Completed and Draft form tabs, include a tab that displays all data (complete and incomplete)_:
* Replaced the term "Forms" with "Questionnaires"
* To match the new “Subjects” box title, replaced the “Forms” box title with tabs: "Questionnaires" (all data), "Completed", "Drafts"
- [x] [CARDS-1442](https://phenotips.atlassian.net/browse/CARDS-1442) _The side bar should list by default "Patients" and "Questionnaires" instead of "Subjects" and "Forms"_
- [x] [CARDS-1445](https://phenotips.atlassian.net/browse/CARDS-1445) _Reword the (+) button titles (on Subject chart, Subjects, Forms) and the default menu items for the Dashboard's (+) button_

marta- committed:
CARDS-1415 / CARDS-1434: "Subjects" box improvements
* Removed the “Subjects” title
* Move the tabs up instead of the title, and used title font
* Make all subject type labels plural
* To match the new “Subjects” box title, replaced the “Forms” box title with the tabs (re-labeled as “Completed forms” and 'Drafts”)

sashaandjic reviewed (approved):
awesome - it feels so natural that a few times I had to go back to compare it to the previous look

k-pace reviewed (approved):
Extremely minor so approving but can we capitalize the f in forms?

marta- committed:
CARDS-1415 / CARDS-1434: "Subjects" box improvements

Addressing review comment: Capitalized tab label

marta- commented:
> Extremely minor so approving but can we capitalize the f in forms?

Done in 14ff90a .

sashaandjic reviewed (approved):
![image](https://user-images.githubusercontent.com/23243244/137423502-b618c7f6-9f39-4547-9cb6-53856467ec00.png)
![image](https://user-images.githubusercontent.com/23243244/137423557-b53463a0-349d-4e76-a81e-8ab7909e3171.png)

sashaandjic commented:
QUESTION: Should `Tumor regions` be `Tumor Regions`? 
![image](https://user-images.githubusercontent.com/23243244/137423841-0c7a51d0-9dfe-4a06-829f-a2674e6bded5.png)

k-pace commented:
> QUESTION: Should `Tumor regions` be `Tumor Regions`? 
> 
> ![image](https://user-images.githubusercontent.com/23243244/137423841-0c7a51d0-9dfe-4a06-829f-a2674e6bded5.png)
> 
> 

Yes, I think so

sdumitriu commented:
> Should `Tumor regions` be `Tumor Regions`?

Should this change be done globally, i.e. always have `Tumor Region` displayed instead of `Tumor region`? Or just in this tab label?

sdumitriu reviewed (changes requested):
Bug: switching from Completed Forms to Drafts does not work.

marta- committed:
CARDS-1415 / CARDS-1434: "Subjects" box improvements
Bugfix: cannot switching to "Drafts" tab has no effect

marta- commented:
> Bug: switching from Completed Forms to Drafts does not work.

Fixed in a2f509c .

marta- committed:
CARDS-1415 / CARDS-1438: "Forms" box improvements
CARDS-1370: In addition to Completed and Draft form tabs, include a tab that displays all data (complete and incomplete)

sashaandjic reviewed (approved):
tested - problem when switching tabs fixed

marta- committed:
CARDS-1415/CARDS-1434: "Subjects" box improvements
* Added a "Subject list label" property to subject type nodes, to be used as a title for subject listings
* In the hierarchy/breadcrumbs displayed on the subject chart, replaced "Subjects" label of the first link with the subject list label of the subject type of the root subject
* Specified the "subject list label" property for currently configured subject types

marta- committed:
CARDS-1415/CARDS-1442: The side bar should list by default "Patients" and "Questionnaires" instead of "Subjects" and "Forms"

sashaandjic reviewed (approved):
**tested**

![image](https://user-images.githubusercontent.com/23243244/137613777-6e17d622-d2ce-4604-909c-05b013eecec8.png)
![image](https://user-images.githubusercontent.com/23243244/137614011-ff9b7bee-4ded-403b-affa-0ca231eaa364.png)
![image](https://user-images.githubusercontent.com/23243244/137614057-12b4dc35-1723-44d1-9d36-12d743d198c2.png)
![image](https://user-images.githubusercontent.com/23243244/137614121-b7298347-f0db-4213-bfcf-269fc33d797c.png)

marta- committed:
CARDS-1415/CARDS-1445: Reword the (+) button titles (on Subject chart, Subjects, Forms) and the default menu items for the Dashboard's (+) button

k-pace commented:
I might say "Add a questionnaire", but maybe we're just getting into things that don't matter. My rationale - filling out to me is more of a patient / basic user function or done through edit mode, whereas the + to me indicates doing something new

sashaandjic commented:
Not sure if I like this one. Somehow the title "New" seems to create a confusion here. With the reworded options I would remove it.
 Also, "Fill out a questionnaire" can be interpreted also as filling out incomplete one. Users may get confused here.
![image](https://user-images.githubusercontent.com/23243244/137660791-ebae435f-d944-4adf-8965-6de3de23dfc5.png)

Should "form" be replaced here with "questionnaire"?
![image](https://user-images.githubusercontent.com/23243244/137660579-2b5ecff6-2968-4728-8051-b462d4e4ec6d.png)

sashaandjic reviewed (approved):
tested - looks awesome

sashaandjic commented:
We can completely eliminate word "Subject" if we implement this [IDEA task](https://phenotips.atlassian.net/browse/CARDS-1447) .

marta- committed:
CARDS-500/CARDS-1132: Display the configured statistics on the stats dashboard
Use the subjectListLabel (if it exists) for the y axis
